library/src/G4CMPProcessUtils.cc

```diff
@@ -504,5 +504,5 @@ G4CMPProcessUtils::ChargeCarrierTimeStep(G4double mach, G4double l0) const {
   const G4double velLong = theLattice->GetSoundSpeed();
 
   const G4double tstep = 3.*l0/velLong;
-  return (mach<1.) ? tstep : tstep*mach/((mach-1)*(mach-1)*(mach-1));
+  return (mach<1.) ? tstep : tstep*mach*mach/((mach-1)*(mach-1)*(mach-1));
```

Comment:
Add a comment showing the derivation that gets the correct "extra" factor of mach in the numerator.

Reply:
it's done!